spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigUtils.java

```diff
@@ -106,15 +112,47 @@ static Map<String, String> getConfigMapData(KubernetesClient client, String name
 		return configMap.getData();
 	}
 
-	/**
-	 * return decoded data from a secret within a namespace.
-	 */
-	static Map<String, Object> dataFromSecret(Secret secret, String namespace) {
-		LOG.debug("reading secret with name : " + secret.getMetadata().getName() + " in namespace : " + namespace);
-		return secretData(secret.getData());
+	static Map<String, Object> secretDataByName(KubernetesClient client, String namespace, String name) {
+		LOG.debug("Loading Secret with name '" + name + "' in namespace '" + namespace + "'");
+		Secret secret = client.secrets().inNamespace(namespace).withName(name).get();
+
+		if (secret == null) {
+			LOG.warn("secret with name : '" + name + "' not present in namespace : '" + namespace + "'");
+			return Collections.emptyMap();
+		}
+
+		return decodeData(secret.getData());
+	}
+
+	static Map.Entry<String, Map<String, Object>> secretDataByLabels(KubernetesClient client, String namespace,
```

Comment:
read secrets by labels, returning the data they hold + the names concatenated (this will be used as the property source name)